main.go

```diff
@@ -194,39 +201,92 @@ func fetchOrCreateTestPlan(ctx context.Context, cfg config.Config, files []strin
 	}
 
 	// If the cache is empty, create a new plan.
+	params := createRequestParam(cfg, files, *apiClient, testRunner)
+	testPlan, err := apiClient.CreateTestPlan(ctx, cfg.SuiteSlug, params)
+
+	if err != nil {
+		// Didn't exceed context deadline? Must have been some kind of error that
+		// means we should return error to main function and abort.
+		if !errors.Is(err, context.DeadlineExceeded) {
+			return plan.TestPlan{}, err
+		}
+		// Create the fallback plan
+		fmt.Println("Could not fetch plan from server, using fallback mode. Your build may take longer than usual.")
+		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
+	}
+
+	// The server can return an "error" plan indicated by an empty task list (i.e. `{"tasks": {}}`).
+	// In this case, we should create a fallback plan.
+	if len(testPlan.Tasks) == 0 {
+		fmt.Println("Test splitter server returned an error, using fallback mode. Your build may take longer than usual.")
+		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
+	}
+
+	return testPlan, nil
+}
```

Comment:
It looks like a lot of changes here, but this is an existing code. The only changes are the extracting params creation into a separate function.